aten/src/ATen/native/PadNd.cpp

```diff
@@ -137,63 +137,63 @@ Tensor _pad_circular(const Tensor &self, IntArrayRef padding) {
     const auto pad_r = padding[2*i + 1];
 
     if (pad_l > 0) {
-      out_slice = out.slice(dim, 0, pad_l);
-      in_slice = out.slice(dim,
-                           out_shape[dim] - pad_l - std::max(pad_r, zero),
-                           out_shape[dim] - std::max(pad_r, zero));
+      out_slice = out.slice_symint(dim, 0, pad_l);
+      in_slice = out.slice_symint(dim,
+                           out_shape[dim] - pad_l - pad_r.max(zero),
+                           out_shape[dim] - pad_r.max(zero));
       out_slice.copy_(in_slice);
     }
 
     if (pad_r > 0) {
-      out_slice = out.slice(dim, out_shape[dim] - pad_r, out_shape[dim]);
-      in_slice = out.slice(dim, std::max(pad_l, zero), std::max(pad_l, zero) + pad_r);
+      out_slice = out.slice_symint(dim, out_shape[dim] - pad_r, out_shape[dim]);
+      in_slice = out.slice_symint(dim, pad_l.max(zero), pad_l.max(zero) + pad_r);
```

Comment:
We should have a SymInt specialization for std::max

Reply:
I added one last week but ended up removing it.

From the [docs](https://en.cppreference.com/w/cpp/algorithm/max), it looks like `std::max/min()` already have a template that just calls into `operator<` and `operator>`. So adding a special handling for min/max on symints in C++ seemed redundant to me, but lmk if you think that's wrong. We still have min/max support in python.

Reply:
ok, I am fine with this resolution